src/app/qgsmaptoolsplitfeatures.cpp

```diff
@@ -93,33 +93,33 @@ void QgsMapToolSplitFeatures::cadCanvasReleaseEvent( QgsMapMouseEvent *e )
     //bring up dialog if a split was not possible (polygon) or only done once (line)
     int topologicalEditing = QgsProject::instance()->topologicalEditing();
     vlayer->beginEditCommand( tr( "Features split" ) );
-    int returnCode = vlayer->splitFeatures( points(), topologicalEditing );
+    QgsGeometry::OperationResult returnCode = vlayer->splitFeatures( points(), topologicalEditing );
     vlayer->endEditCommand();
-    if ( returnCode == 4 )
+    if ( returnCode == QgsGeometry::OperationResult::NothingHappened )
     {
       QgisApp::instance()->messageBar()->pushMessage(
         tr( "No features were split" ),
         tr( "If there are selected features, the split tool only applies to those. If you would like to split all features under the split line, clear the selection." ),
         QgsMessageBar::WARNING,
         QgisApp::instance()->messageTimeout() );
     }
-    else if ( returnCode == 3 )
+    else if ( returnCode == QgsGeometry::OperationResult::GeometryEngineError )
```

Comment:
Not sure if you do it consciously or not. `QgsGeometry::GeometryEngineError` is also valid, I think this kind of usage is more widespread in the code, but I don't think we have a policy.

Reply:
@m-kuhn not much in this PR is conscious ;) I'm only trying to return enums instead of `int` ... btw, what do you mean by 'is also valid' ? That "Cut edges detected." cannot be the cause for returning `QgsGeometry::GeometryEngineError` ? I'll check what the original algorithm returns, I believe at the end it is geos.

Reply:
Only that `QgsGeometry::OperationResult::GeometryEngineError` and `QgsGeometry::GeometryEngineError` are equivalent

Reply:
ah, yes, that was conscious, I find it more readable because you always know from which enum it comes from and auto-completion only show relevant enum items, but what I'd really love to see would be using scoped enums wherever possible.

Reply:
I agree with @elpaso here - this will also ease the transition to C++11 enum classes (when sip has support for them!)